static/js/components/widgets/SelectField.tsx

```diff
@@ -31,23 +34,30 @@ export default function SelectField(props: Props): JSX.Element {
     [name, multiple]
   )
 
+  const getSelectOption = (value: string) => {
+    const selectOption = selectOptions.filter(
+      option => option.value === value
+    )[0]
+    return { label: selectOption ? selectOption.label : value, value: value }
+  }
+
   let selected
   if (multiple) {
     if (!Array.isArray(value)) {
       selected = []
     } else {
-      selected = value.map(option => ({ label: option, value: option }))
+      selected = value.map(option => getSelectOption(option))
```

Comment:
ok so if I'm understanding this right, we need to get options that have a `value` prop in the `value` array - would it maybe make sense to do something like this instead of the `getSelectedOption` thing?

```ts
selected = selectOptions.filter(option => {
  value.includes(option.value)
})
```

I'm not sure if I'm understanding all of what this is doing though! so maybe it's more complicated than that. In any case, might be good to throw in a comment to explain what's up.

Reply:
The original code will return a `{label, value}` `Option` object even if that option is not included in the select form field.  I'm not sure if that's something we need to continue supporting but felt I shouldn't take it away in this PR if that's the way it's working now.   That's why I wrote the `getSelectOption` function to return the existing `Option` if it exists, otherwise generate a new one based on the value (`return { label: selectOption ? selectOption.label : value, value: value }`).

Reply:
Added a comment and simplified the function a bit.

Reply:
@aliceriot ready for another look